addons/website/models/website.py

```diff
@@ -404,7 +404,8 @@ def get_unique_key(self, string, template_module=False):
         key_copy = string
         inc = 0
         domain_static = self.get_current_website().website_domain()
-        while self.env['website.page'].with_context(active_test=False).sudo().search([('key', '=', key_copy)] + domain_static):
+        while self.env['website.page'].with_context(active_test=False).sudo().search([('key', '=', key_copy)] + domain_static)\
+                or self.env['ir.ui.view'].with_context(active_test=False).sudo().search([('key', '=', key_copy)] + domain_static):
```

Comment:
```suggestion
        while self.env['ir.ui.view'].with_context(active_test=False).sudo().search([('key', '=', key_copy)] + domain_static):
```
Replacing `website.page` by `ir.ui.view` was enough no?
Seems like the first condition is a subset of second condition which will involves the same check a second time, as `key` is a property of `ir.ui.view`, not `website.page` (`inherits`)